scripts/generate_abi.py

```diff
@@ -0,0 +1,96 @@
+import json
+import tempfile
+from pathlib import Path
+from typing import Any
+
+import click
+from psutil import Popen
+
+_COMPILE_COMMAND = """
+git clone --no-checkout git@github.com:beamer-bridge/beamer.git {beamer_path} &&
+cd {beamer_path} &&
+git checkout {commit_id} &&
+ape compile
+"""
+
+
+def _get_contract_commit_ids(deployment_path: Path) -> dict[str, set[str]]:
+    file_contract_commits: dict[str, set[str]] = {}
+    for file_path in deployment_path.glob("*.deployment.json"):
+        with open(file_path, "r") as fp:
+            data = json.load(fp)
+        data.pop("deployer")
+        for contract_name, contract_info in data.items():
+            if contract_name not in file_contract_commits:
+                file_contract_commits[contract_name] = set()
+            file_contract_commits[contract_name].add(contract_info["beamer_commit"])
+    return file_contract_commits
+
+
+def _compile_abis(commit_id: str, beamer_path: str) -> None:
+    command = _COMPILE_COMMAND.format(beamer_path=beamer_path, commit_id=commit_id).replace(
+        "\n", ""
+    )
+    Popen(command, shell=True).wait()
+
+
+def _get_contract_abis(
+    generated_abis: dict[str, dict[str, Any]]
+) -> dict[str, list[dict[str, Any]]]:
+    contract_abis: dict[str, list[dict[str, Any]]] = {}
+    for abis in generated_abis.values():
+        for contract_name, abi in abis.items():
+            if contract_name not in contract_abis:
+                contract_abis[contract_name] = []
+            contract_abis[contract_name].append(abi)
+    return contract_abis
+
+
+def _merge_abis(contract_abis: dict[str, list[Any]]) -> dict[str, dict[str, Any]]:
+    merged_abis = {}
+
+    for contract_name, abis in contract_abis.items():
+        if all(abi == abis[0] for abi in abis):
+            merged_abis[contract_name] = abis[0]
```

Comment:
ABIs are just strings, right? So we could use the same approach as for commits and store them into a set.
And here we just need to check whether the size of the set is greater than 1.

Actually, if we move this check to `_generate_abis`, we don't even need this function.

Reply:
it is a dict and i cant hash it

Reply:
i am returning everything ape provides as abi

Reply:
now i got what you mean, changed it. i guess now it is better